the-collab-lab/tcl-30-whats-near-me
# [TCL30-14] Handle the user stopping sharing of their location

## Description

This PR allows the user to stop sharing their location with the app to conserve resources on their device and protect their privacy. It makes sure that a "Stop Sharing Location" button is rendered on the screen once the user has shared their device location. It also ensures that when the user clicks the ``Stop Sharing`` prompt, the user location is no longer watched by the app, and the app returns to the default state.

## Related Issue

Closes [TCL30-14](https://the-collab-lab.atlassian.net/browse/TCL30-14)


## Acceptance Criteria

-  Once the user shared their location, they should have available 'Stop sharing location' button 
-  Once the user shared their location, the 'Share Location' button should become disabled 
-  Once  the user stopped sharing their location, the user marker should be removed from the map,  the handler watching their position should be unregistered, and the 'Share Location' button should return to its default state, allowing the user to start sharing location again

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before

 iPhone 6/7/8 Map View -BEFORE   |   Map View -BEFORE
:----------------------------------:|:-------------------------------------:
![image](https://user-images.githubusercontent.com/16920088/128629228-b65c04b4-3d65-44d3-a434-dd0cccc0012f.png) | ![image](https://user-images.githubusercontent.com/16920088/128629235-324c3efd-d6c5-4061-8b63-073b5d7aea1f.png)








### After
 iPhone 6/7/8 Map View -AFTER   |   Map View -AFTER 
:----------------------------------:|:-------------------------------------:
![image](https://user-images.githubusercontent.com/16920088/128641805-5aa76134-8e9a-4cf2-b18d-43504d38e9de.png) | ![image](https://user-images.githubusercontent.com/16920088/128641814-0136aec1-a7fc-49bf-be37-e8bf6f31e7ea.png)










## Testing Steps / QA Criteria

- From your terminal, pull down this branch ``ku-mm-handle-stop-sharing``
- Run ``npm start`` to launch the app 
- Make sure that once you shared your location with the app, you have  a 'Stop sharing location' button available, and  
the 'Share Location' button is disabled
- Make sure that once you stop sharing your location, the marker designating your location is removed from the map, the ``Share Location`` button returns to its default state, and the handler watching your position is unregistered.  There is a ``console.log`` statement to verify that the location watch handler is unregistered and your location is no longer returned by the browser. Check the console to confirm the messages.
